keel-core/src/main/kotlin/com/netflix/spinnaker/keel/verification/VerificationRunner.kt

```diff
@@ -0,0 +1,44 @@
+package com.netflix.spinnaker.keel.verification
+
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.plugins.VerificationEvaluator
+import com.netflix.spinnaker.keel.api.verification.VerificationRepository
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.RUNNING
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class VerificationRunner(
+  private val verificationRepository: VerificationRepository,
+  private val evaluators: List<VerificationEvaluator<*>>
+) {
+  fun runVerificationsFor(environment: Environment, artifact: DeliveryArtifact, version: String) {
+    val statuses = environment
+      .verifyWith
+      .map {
+        it to verificationRepository
+          .getState(it, environment, artifact, version)
+          ?.status
+      }
+
+    if (statuses.any { (_, status) -> status == RUNNING }) {
+      log.debug("Verification already running for {}", environment.name)
+      return
+    }
```

Comment:
I'm thinking we probably need to call another method on `VerificationEvaluator` here to see if it's done yet and store the updated state. Still thinking about how that should look.